wrappers/react/src/hotEditor.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react';
+import Handsontable from 'handsontable/base';
+import { HotEditorHooks } from './types';
+import { superBound } from "./helpers";
+
+/**
+ * Create a class to be passed to the Handsontable's settings.
+ *
+ * @param {React.RefObject<HotEditorHooks>} hooksRef Reference to component-based editor overridden hooks object.
+ * @param {React.RefObject} instanceRef Reference to Handsontable-native custom editor class instance.
+ * @returns {Function} A class to be passed to the Handsontable editor settings.
+ */
+export function makeEditorClass(hooksRef: React.MutableRefObject<HotEditorHooks>, instanceRef: React.MutableRefObject<Handsontable.editors.BaseEditor>): typeof Handsontable.editors.BaseEditor {
+  return class CustomEditor extends Handsontable.editors.BaseEditor implements Handsontable.editors.BaseEditor {
+    constructor(hotInstance: Handsontable.Core) {
+      super(hotInstance);
+      instanceRef.current = this;
+
+      Object.getOwnPropertyNames(Handsontable.editors.BaseEditor.prototype).forEach(propName => {
+        if (propName === 'constructor') {
+          return;
+        }
+
+        const baseMethod = Handsontable.editors.BaseEditor.prototype[propName];
+        CustomEditor.prototype[propName] = function (...args) {
+          if (hooksRef.current?.[propName]) {
+            return hooksRef.current[propName].call(this, ...args);
+          } else {
+            return baseMethod.call(this, ...args);
+          }
+        }.bind(this);
+      });
+    }
+
+    focus() {
+    }
+
+    getValue() {
+    }
+
+    setValue() {
+    }
+
+    open() {
+    }
+
+    close() {
+    }
+  }
+}
+
+/**
+ * Context to provide Handsontable-native custom editor class instance to overridden hooks object.
+ */
+const EditorContext = React.createContext<React.RefObject<Handsontable.editors.BaseEditor>>(undefined);
+
+interface EditorContextProviderProps {
+  classInstanceRef: React.RefObject<Handsontable.editors.BaseEditor>
+  children: React.ReactNode
+}
+
+/**
+ * Provider of the context that exposes Handsontable-native editor instance for custom editor components.
+ *
+ * @param {React.RefObject} classInstanceRef  Reference to Handsontable-native editor instance.
+ */
+export const EditorContextProvider: React.FC<EditorContextProviderProps> = ({ classInstanceRef, children }) => {
+  return <EditorContext.Provider value={classInstanceRef}>
+    {children}
+  </EditorContext.Provider>
+}
+
+/**
+ * Helper function that wraps React.ForwardRef for ease of creating custom component-based editors.
+ *
+ * @param {Function} render The editor component function.
+ */
+export function hotEditor<P>(render: React.ForwardRefRenderFunction<HotEditorHooks, P>) {
```

Comment:
lets remove this helper function that wraps all together, its another level of friction from the end user we want to avoid.